### PR TITLE
TA updates

### PR DESCRIPTION
**Pull request by jenswi-linaro**
Gives an overview of REE FS TAs and ldelf.

### CODE REVIEW

File: architecture/trusted_applications.rst [hunk 1]
```diff
@@ -81,6 +81,117 @@ filesystem, and ``tee-supplicant`` will take care of passing them to be checked
 and loaded by the Secure World OP-TEE core. Note that this type of TA isn't
 encrypted.
 
+REE filesystem TAs come in two formats, the legacy TA and the bootstrap TA.
```
Comment (jforissier) [inline architecture/trusted_applications.rst]: Maybe say that `scripts/sign.py` uses the boostrap TA format since version 3.7.0?

Comment (jenswi-linaro) [reply architecture/trusted_applications.rst]: I'll fix.

**Comment by jenswi-linaro** [comment]
"Update REE filesystem TA section" fixed up according to comment, applied tags.

File: architecture/trusted_applications.rst [hunk 2]
```diff
@@ -99,6 +211,40 @@ with the command:
     $ xtest --install-ta
 
 
+Loading and preparing TA for exeuction
```
Comment (jforissier) [inline architecture/trusted_applications.rst]: `s/exeuction/execution/` and in the commit subject too

**Comment by jenswi-linaro** [comment]
Comment addressed.

File: architecture/trusted_applications.rst [hunk 3]
```diff
@@ -81,6 +81,118 @@ filesystem, and ``tee-supplicant`` will take care of passing them to be checked
...
+
+    hash = H(<struct shdr> || <stripped ELF>)
+    signature = RSA-Sign(hash)
+    legacy_binary = <struct shdr> || <hash> || <signature> || <stripped ELF>
+
+Bootstrap TA binary is formatted as:
+
+.. code-block:: none
+
+    hash = H(<struct shdr> || <struct shdr_bootstrap_ta> || <stripped ELF>)
```
Comment (jbech-linaro) [inline architecture/trusted_applications.rst]: Which hash algorithm? `SHA-256`? It's important information and should be mentioned somewhere.

File: architecture/trusted_applications.rst [hunk 4]
```diff
@@ -81,6 +81,118 @@ filesystem, and ``tee-supplicant`` will take care of passing them to be checked
...
+    signature = RSA-Sign(<hash>)
```
Comment (jbech-linaro) [inline architecture/trusted_applications.rst]: Which RSA algorithm? `RSASSA_PKCS1_V1_5_SHA256`?

File: architecture/trusted_applications.rst [hunk 5]
```diff
@@ -81,6 +81,118 @@ filesystem, and ``tee-supplicant`` will take care of passing them to be checked
...
+                       <struct shdr_bootstrap_ta> || <stripped ELF>
+
+A REE TA is loaded into shared memory using a series or RPC in
+:ref:`load_ree_ta`. The payload memory is allocated via TEE-supplicant and
+later freed when the TA has been loaded into secure memory in
+:ref:`free_appl_shm`.
+
+.. _load_ree_ta:
+
+.. figure:: ../images/trusted_applications/load_ree_ta.png
```
Comment (jbech-linaro) [inline architecture/trusted_applications.rst]: I really like this sequence diagram!

**Comment by jenswi-linaro** [comment]
Update with algorithm

**Comment by jbech-linaro** [comment]
`Reviewed-by: Joakim Bech <joakim.bech@linaro.org>`

**Comment by jenswi-linaro** [comment]
Tag applied

**Pull request merged**